pennylane/_qubit_device.py

```diff
@@ -960,7 +971,27 @@ def var(self, observable, shot_range=None, bin_size=None):
         samples = self.sample(observable, shot_range=shot_range, bin_size=bin_size)
         return np.squeeze(np.var(samples, axis=0))
 
-    def sample(self, observable, shot_range=None, bin_size=None):
+    def _samples_to_counts(self, samples):
+        """Group the obtained samples into a dictionary.
+
+        Example:
+            >>> samples
+            tensor([[0, 0, 1],
+                    [0, 0, 1],
+                    [1, 1, 1]], requires_grad=True)
+            >>> self._samples_to_counts(samples)
+             {'111':1, '001':2}
+        """
+        try:
+            # Express the states as strings
+            samples = ["".join(self._asarray(sample, dtype=str)) for sample in samples]
+        except TypeError:
+            # Evaluation of an observable
```

Comment:
Why exactly does a `TypeError` arise in this case? Maybe worth extending the comment a bit.